src/main/java/com/microsoft/spring/data/gremlin/config/GremlinConfigurationSupport.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.config;
+
+import com.microsoft.spring.data.gremlin.mapping.GremlinMappingContext;
+import lombok.SneakyThrows;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.type.filter.AnnotationTypeFilter;
+import org.springframework.data.annotation.Persistent;
+import org.springframework.data.mapping.context.MappingContext;
+import org.springframework.data.mapping.context.MappingContextIsNewStrategyFactory;
+import org.springframework.data.mapping.context.PersistentEntities;
+import org.springframework.data.support.CachingIsNewStrategyFactory;
+import org.springframework.data.support.IsNewStrategyFactory;
+import org.springframework.lang.NonNull;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.StringUtils;
+
+import java.util.*;
+
+public abstract class GremlinConfigurationSupport {
+
+    protected Collection<String> getMappingBasePackages() {
+        final Package basePackage = this.getClass().getPackage();
```

Comment:
Can we move this `basePackage` to a `private static final` class level variable to avoid multiple process?